src/App.js

```diff
@@ -1,19 +1,49 @@
 import React from 'react';
 import './App.css';
 
+import todos from './api/todos';
 import users from './api/users';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+import { TodoList } from './components/TodoList';
+import { NewTodoForm } from './components/NewTodoForm';
+
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId),
+}));
+
+class App extends React.PureComponent {
+  state = {
+    todos: preparedTodos,
+  };
+
+  addTodo = (user, title) => {
+    const newTodo = {
+      id: Math.random() * 1000,
+      title,
+      completed: false,
+      user: {
+        name: user,
+      },
```

Comment:
You may just pass the name, without an object

Reply:
Small update:
Actually, we don't need to pass the 'name' only 'userId'.
Each TODO in our list looks like this:
```
 {
    userId: 1,
    id: 1,
    title: 'delectus aut autem',
    completed: false,
  },
```
So we should pass 'userId' to our new TODO-object, as task description says:

> each TODO should have a title and userId selected from a list of given users

Also, you may pass the 'Name' of the user to the TodoList to make it looks prettier, but not to add it to the new 'todo' object and it's not essential for this task